src/vm/field.cpp

```diff
@@ -886,6 +886,13 @@ TypeHandle FieldDesc::GetExactFieldType(TypeHandle owner)
         PCCOR_SIGNATURE pSig;
         DWORD cSig;
         GetSig(&pSig, &cSig);
+
+        // Point past the calling convention
+        BYTE * pOldPtr = (BYTE*) pSig;
+        unsigned callConv = CorSigUncompressData(pSig);
+        _ASSERTE(callConv == IMAGE_CEE_CS_CALLCONV_FIELD);
```

Comment:
It would be better to call GetCallingConv on the SigPointer, e.g. like what is done here: https://github.com/dotnet/coreclr/blob/7ced6b540e5c2fd7ed49661dc0b182d65a413949/src/vm/siginfo.cpp#L2925